Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,419 @@
+                    std::ofstream::out | std::ofstream::app);
+                // write header row
+                ofs << "#";
+                ofs << "[1]step()";
+                ofs << "[2]time(s)";
+                ofs << "[3]xmin()";
+                ofs << m_sep;
+                ofs << "[4]xmax()";
+                ofs << m_sep;
+                ofs << "[5]ymin()";
+                ofs << m_sep;
+                ofs << "[6]ymax()";
+                ofs << m_sep;
+                ofs << "[7]zmin()";
+                ofs << m_sep;
+                ofs << "[8]zmax()";
+                ofs << m_sep;
+                ofs << "[9]pxmin()";
+                ofs << m_sep;
+                ofs << "[10]pxmax()";
+                ofs << m_sep;
+                ofs << "[11]pymin()";
+                ofs << m_sep;
+                ofs << "[12]pymax()";
+                ofs << m_sep;
+                ofs << "[13]pzmin()";
+                ofs << m_sep;
+                ofs << "[14]pzmax()";
+                ofs << m_sep;
+                ofs << "[15]gmin()";
+                ofs << m_sep;
+                ofs << "[16]gmax()";
+                ofs << m_sep;
+                ofs << "[17]wmin()";
+                ofs << m_sep;
+                ofs << "[18]wmax()";
+                if (myspc.has_breit_wheeler() || myspc.has_quantum_sync())
+                {
+                    ofs << m_sep;
+                    ofs << "[19]chimin()";
+                    ofs << m_sep;
+                    ofs << "[20]chimax()";
+                }
+                ofs << std::endl;
+                // close file
+                ofs.close();
+            }
+        }
+    }
+}
+// end constructor
+
+// function that computes extrema
+void ParticleExtrema::ComputeDiags (int step)
+{
+
+    // Judge if the diags should be done
+    if (!m_intervals.contains(step+1)) { return; }
+
+    // get MultiParticleContainer class object
+    auto & mypc = WarpX::GetInstance().GetPartContainer();
+
+    // get number of level (int)
+    auto level_number = WarpX::GetInstance().finestLevel();
+
+    // get number of species (int)
+    auto nSpecies = mypc.nSpecies();
+
+    // get species names (std::vector<std::string>)
+    auto species_names = mypc.GetSpeciesNames();
+
+    // inverse of speed of light squared
+    Real constexpr inv_c2 = 1.0 / (PhysConst::c * PhysConst::c);
+
+    // If 2D-XZ, p.pos(1) is z, rather than p.pos(2).
+#if (defined WARPX_DIM_3D)
+    int const index_z = 2;
+#elif (defined WARPX_DIM_XZ || defined WARPX_DIM_RZ)
+    int const index_z = 1;
+#endif
+
+    // loop over species
+    for (int i_s = 0; i_s < nSpecies; ++i_s)
+    {
+
+        // only chosen species does
+        if (species_names[i_s] != m_species_name) { continue; }
+
+        // get WarpXParticleContainer class object
+        auto & myspc = mypc.GetParticleContainer(i_s);
+
+        // get mass (Real)
+        auto m = myspc.getMass();
```

Comment:
```suggestion
        const auto m = myspc.getMass();
```
I think that this could be `const`